eva/third_party/huggingface/create.py

```diff
@@ -0,0 +1,203 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+from typing import Dict, List, Type, Union
+
+import cv2
+import numpy as np
+from PIL import Image, ImageDraw
+from transformers import Pipeline, pipeline
+
+from eva.catalog.catalog_type import ColumnType, NdArrayType
+from eva.catalog.models.udf_io_catalog import UdfIOCatalogEntry
+from eva.catalog.models.udf_metadata_catalog import UdfMetadataCatalogEntry
+from eva.third_party.huggingface.model import HFInputTypes
+
+"""
+We currently support the following tasks from HuggingFace.
+"""
+SUPPORTED_TASKS = {
+    "audio-classification": HFInputTypes.AUDIO,
+    "automatic-speech-recognition": HFInputTypes.AUDIO,
+    "text-classification": HFInputTypes.TEXT,
+    "summarization": HFInputTypes.TEXT,
+    "translation": HFInputTypes.TEXT,
+    "text2text-generation": HFInputTypes.TEXT,
+    "text-generation": HFInputTypes.TEXT,
+    "image-classification": HFInputTypes.IMAGE,
+    "image-segmentation": HFInputTypes.IMAGE,
+    "image-to-text": HFInputTypes.IMAGE,
+    "object-detection": HFInputTypes.IMAGE,
+    "depth-estimation": HFInputTypes.IMAGE,
+    "video-classification": HFInputTypes.VIDEO,
+}
+
+
+def run_pipe_through_text(pipe: Pipeline):
```

Comment:
How about we have different functions to create the required data and decouple the `pipe(data)` call? I feel we will be able to clean it further.

Reply:
I initially did it that way, but the reason for not going ahead was for video. I would need a temp file for video and this was the cleanest way of dealing with videos.